lib/vmdb/settings_walker.rb

```diff
@@ -20,13 +20,14 @@ def walk(settings, path = [], &block)
           key_path = path.dup << key
 
           yield key, value, key_path, settings
+          next if key == settings || value == settings
```

Comment:
How would the key equal the settings object? And would that even lead to this recursion behavior?

Reply:
ok. you win. I got in examples on all 3 of these cases.